static/js/context-menu.js

```diff
@@ -60,7 +62,50 @@ const ContextMenu = {
     switch (e.detail.action) {
       case 'edit':
         this.nameInput.value = e.detail.thingName;
+        this.thingType.innerHTML = '';
         this.thingIcon.style.backgroundImage = `url("${e.detail.thingIcon}")`;
+
+        for (const capability of e.detail.capabilities) {
+          const option = document.createElement('option');
+          option.value = capability;
+
+          if (e.detail.selectedCapability === capability) {
+            option.selected = true;
+          }
+
+          switch (capability) {
+            case 'OnOffSwitch':
+              option.innerText = 'On/Off Switch';
+              break;
+            case 'MultiLevelSwitch':
+              option.innerText = 'Multi Level Switch';
+              break;
+            case 'ColorControl':
+              option.innerText = 'Color Control';
+              break;
+            case 'EnergyMonitor':
+              option.innerText = 'Energy Monitor';
+              break;
+            case 'BinarySensor':
+              option.innerText = 'Binary Sensor';
+              break;
+            case 'MultiLevelSensor':
+              option.innerText = 'Multi Level Sensor';
+              break;
+            case 'SmartPlug':
+              option.innerText = 'Smart Plug';
+              break;
+            case 'Light':
+              option.innerText = 'Light';
+              break;
+            default:
+              option.innerText = capability;
+              break;
+          }
+
+          this.thingType.appendChild(option);
```

Comment:
Another behavior is that if there are no options, i.e. a typeless unknown thing, the drop-down menu is the kind of subpar empty drop-down (which encourages clicking but then shows up with nothing) instead of having a fallback "Thing" option. This also applies to the add thing menu

<img width="584" alt="screen shot 2018-06-28 at 12 41 36 pm" src="https://user-images.githubusercontent.com/2776089/42048244-b00baf3c-7ad0-11e8-94ec-aa03013bb721.png">

Reply:
Yeah, I left that intentionally. I'm going to be adding custom icon selection after this is merged, so there will _always_ be something in the dropdown.